applications/dashboard/controllers/class.rolecontroller.php

```diff
@@ -157,6 +175,41 @@ public function edit($RoleID = false) {
                 $this->Form->setFormValue('PersonalInfo', false);
             }
 
+            // If the form didn't have category permissions, they'll need to be added before saving.
```

Comment:
Again, we want the setting to not re-add the categories or else the optimization is lost. When saving a role's permissions I just want to save two rows in GDN_Permission:

1. The global permission row.
2. The default category permissions row.